gammapy/image/maps.py

```diff
@@ -416,13 +417,18 @@ def lookup_max(self, region=None):
 
     def solid_angle(self):
         """
-        Solid angle image
+        Solid angle image (2-dim `astropy.units.Quantity` in `sr`).
         """
         coordinates = self.coordinates(mode='edges')
-        xsky = coordinates.data.lon
-        ysky = coordinates.data.lat
-        omega = np.abs(np.diff(xsky, axis=1)[1:, :] * np.diff(ysky, axis=0)[:, 1:])
-        return omega.to('sr')
+        lon = coordinates.data.lon.radian
+        lat = coordinates.data.lat.radian
+
+        dx = angular_separation(lon[:, :-1], lat[:, :-1],
+                                    lon[:, 1:], lat[:, :-1])
+
+        dy = angular_separation(lon[:-1, :], lat[:-1, :],
+                                lon[1:, :], lat[1:, :])
+        return Quantity(dx[1:, :] * dy[:, 1:], "sr")
```

Comment:
At first glance `dx` and `dy` look correct to me (but I havn't looked in detail...). But taking again slices of `dx` and `dy` here doesn't seem correct to me (because you already did before). So this line should just read `dx * dy`. This probably caused the test failure.